src/ApimEventProcessor/HttpMessage.cs

```diff
@@ -1,11 +1,58 @@
 using System;
 using System.IO;
+using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Text;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
 
 namespace ApimEventProcessor
 {
+    public class MoesifRequest
+    {
+        public MoesifRequest(JObject request)
+        {
+            eventType = (string) request["event_type"];
+            messageId = (string) request["message-id"];
+            method = (string) request["method"];
+            uri = (string) request["uri"];
+            userId = (string) request["user_id"];
+            companyId = (string) request["company_id"];
```

Comment:
nitpic: Instead of hardcoding strings `company_id`, are we able to reference them as Moesif.Api.Models.EventModel.CompanyId ? Reduces chances of errors, finds errors at compile time, and reduces dependencies if (rarely) the underlying value changes. Maybe put all strings in `Moesif.Api.Models.EventModelConstants` and refer them everywhere.

Reply:
These are the fields we'd read from the xml policy json-like structure and are not referred elsewhere. We could make it as private readonly string to avoid any confusion.